### PR TITLE
Add sysbuild configurations

### PR DESCRIPTION
**Pull request by ktaborowski**
## CI parameters

```yaml
Github_actions:
  #(branch, hash, pull/XXX/head)
  NRF_revision: pull/13153/head

  # Do not change after creating PR
  Create_NRF_PR: false
Jenkins:
  test-sdk-sidewalk: master
```

NRF_revision=main
manifest-pr-skip (remove this line to create PR in NRF)

## Description

JIRA ticket: KRKNWK-18781

## Self review

- [x] There is no commented code.
- [x] There is no TODO/FIXME comments without associated issue ticket.
- [ ] Commits are properly organized
- [ ] Verification
  - [ ] Unit tests were updated to include the change.
  - [ ] Change has been tested. 

## CI parameters

(branch, hash, pull/XXX/head)
NRF_revision=main

### CODE REVIEW

**Comment by ktaborowski** [comment]
based on https://github.com/nrfconnect/sdk-sidewalk/pull/538

**Comment by ktaborowski** [comment]
Manually tested:
* nRF5340 - build, flash, smoke test + dfu test
* nRF52840 and nrf54l15 - build, flash, read mfg

File: samples/sid_end_device/sample.yaml [hunk 1]
```diff
@@ -1,113 +1,63 @@
...
     tags: Sidewalk hello
 
   sample.sidewalk.hello.release:
-    build_only: true
-    platform_allow:
-      - nrf52840dk_nrf52840
-      - nrf5340dk_nrf5340_cpuapp
-      - nrf54l15pdk_nrf54l15_cpuapp
     extra_args:
       CONF_FILE=prj_release.conf
```
Comment (nordicjm) [inline samples/sid_end_device/sample.yaml:23]: would change to `FILE_SUFFIX=release` though note difference in child/parent image and sysbuild:

- child/parent: `FILE_SUFFIX` only applies to main image, you need to repeat it multiple times with image name to apply to other images e.g. `mcuboot_FILE_SUFFIX`, `b0_FILE_SUFFIX`
- sysbuild: `FILE_SUFFIX` applies to **all** images, to only apply to the main image use `<app>_FILE_SUFFIX` where `<app>` is name of the app, generally the folder name

File: samples/sid_end_device/sample.yaml [hunk 2]
```diff
@@ -1,113 +1,63 @@
...
     tags: Sidewalk hello BLE
 
   sample.sidewalk.demo:
-    build_only: true
-    platform_allow:
-      - nrf52840dk_nrf52840
-      - nrf5340dk_nrf5340_cpuapp
-      - nrf54l15pdk_nrf54l15_cpuapp
     extra_args:
       OVERLAY_CONFIG="overlay-demo.conf"
```
Comment (nordicjm) [inline samples/sid_end_device/sample.yaml:42]: change to `EXTRA_CONF_FILE`

File: samples/sid_end_device/sample.yaml [hunk 3]
```diff
@@ -1,113 +1,63 @@
...
 
   sample.sidewalk.demo.ble_only:
-    build_only: true
     platform_allow:
-      - nrf52840dk_nrf52840
-      - nrf5340dk_nrf5340_cpuapp
       - thingy53_nrf5340_cpuapp
-      - nrf54l15pdk_nrf54l15_cpuapp
     extra_args:
       OVERLAY_CONFIG="overlay-demo.conf"
```
Comment (nordicjm) [inline samples/sid_end_device/sample.yaml:51]: etc.

**Comment by totyz** [comment]
todo:
1. simple_bootloader is built using child_image - move to sysbuild
2. lack of DFU *.zip packages

File: samples/sid_end_device/sample.yaml (hunk 1 above)
Comment (ktaborowski) [reply samples/sid_end_device/sample.yaml]: Don't get it, in sysbuild does the file_suffix apply to all images or only to main image?

File: samples/sid_end_device/sample.yaml (hunk 2 above)
Comment (ktaborowski) [reply samples/sid_end_device/sample.yaml]: Is `OVERLAY_CONFIG` deprecated? I see it in many samples and documentation articles

File: samples/sid_end_device/sample.yaml (hunk 1 above)
Comment (nordicjm) [reply samples/sid_end_device/sample.yaml]: sysbuild: `-DFILE_SUFFIX` applies to all, `-Dmcuboot_FILE_SUFFIX` only applies to mcuboot, `-D<app_name>_FILE_SUFFIX` only applies to `<app_name>` image

File: tests/manual/simple_bootloader/CMakeLists.txt [hunk 4]
```diff
@@ -8,8 +8,6 @@ cmake_minimum_required(VERSION 3.20.0)
 
 include($ENV{ZEPHYR_BASE}/../sidewalk/samples/sid_end_device/bootloader_version.cmake)
```
Comment (nordicjm) [inline tests/manual/simple_bootloader/CMakeLists.txt:9]: this is a no

File: tests/manual/simple_bootloader/CMakeLists.txt [hunk 5]
```diff
@@ -8,8 +8,6 @@ cmake_minimum_required(VERSION 3.20.0)
...
 
-set(mcuboot_KCONFIG_ROOT $ENV{ZEPHYR_BASE}/../sidewalk/samples/sid_end_device/child_image/mcuboot/Kconfig.root)
-
 find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
 project(simple_bootloader)
```
Comment (nordicjm) [inline tests/manual/simple_bootloader/CMakeLists.txt:13]: scripts/generate_version_info_overlay.py and samples/sid_end_device/bootloader_version.cmake are also **NO**. Use a VERSION file

**Review by nordicjm — CHANGES_REQUESTED** [review]
Fixup for some non-mentioned issues: https://github.com/ktaborowski/sdk-sidewalk/pull/10

**Comment by ktaborowski** [comment]
> Fixup for some non-mentioned issues: [ktaborowski#10](https://github.com/ktaborowski/sdk-sidewalk/pull/10)

@nordicjm what about ack of DFU *.zip packages? Was it removed pernamently with sysbuild?

**Comment by nordicjm** [comment]
> > Fixup for some non-mentioned issues: [ktaborowski#10](https://github.com/ktaborowski/sdk-sidewalk/pull/10)
> 
> @nordicjm what about ack of DFU *.zip packages? Was it removed pernamently with sysbuild?

That is generated if mcuboot is enabled

**Comment by ktaborowski** [comment]
work in progress...

**Comment by ktaborowski** [comment]
> > > Fixup for some non-mentioned issues: [ktaborowski#10](https://github.com/ktaborowski/sdk-sidewalk/pull/10)
> > 
> > 
> > @nordicjm what about ack of DFU *.zip packages? Was it removed pernamently with sysbuild?
> 
> That is generated if mcuboot is enabled

need to enable some configs, but i found it in other samples, now it should be as before

File: samples/sid_end_device/sysbuild/mcuboot/boards/nrf52840dk_nrf52840.overlay [hunk 6]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/ {
+	chosen {
+		zephyr,code-partition = &boot_partition;
```
Comment (ktaborowski) [inline samples/sid_end_device/sysbuild/mcuboot/boards/nrf52840dk_nrf52840.overlay:9]: @nordicjm I move this change form your fixup commit https://github.com/ktaborowski/sdk-sidewalk/pull/10/commits/36dde823847f56c0875704bde7f47e62651cd94c
Could you explain what it is for?

File: tests/manual/simple_bootloader/CMakeLists.txt (hunk 5 above)
Comment (ktaborowski) [reply tests/manual/simple_bootloader/CMakeLists.txt:11]: Removed the script, but mcuboot can't find the version from file, it's always 0.0.0

File: samples/sid_end_device/Kconfig.sysbuild [hunk 7]
```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (c) 2023 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+config NRF_DEFAULT_BLUETOOTH
+	default y if (("$(BOARD)" = "nrf5340dk_nrf5340_cpuapp") || ("$(BOARD)" = "thingy53_nrf5340_cpuapp"))
```
Comment (nordicjm) [inline samples/sid_end_device/Kconfig.sysbuild]: do not check board names, use Kconfig symbols

File: samples/sid_end_device/Kconfig.sysbuild [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+
+choice BOOTLOADER
+	default BOOTLOADER_MCUBOOT
+endchoice
+
+if BOOTLOADER_MCUBOOT && (("$(BOARD)" = "nrf5340dk_nrf5340_cpuapp") || ("$(BOARD)" = "thingy53_nrf5340_cpuapp"))
```
Comment (nordicjm) [inline samples/sid_end_device/Kconfig.sysbuild]: etc.

File: samples/sid_end_device/sysbuild/mcuboot/boards/nrf52840dk_nrf52840.overlay [hunk 9]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/ {
+	chosen {
+		nordic,pm-ext-flash = &mx25r64;
```
Comment (nordicjm) [inline samples/sid_end_device/sysbuild/mcuboot/boards/nrf52840dk_nrf52840.overlay:10]: this is missing the `zephyr,code-partition` line

File: samples/sid_end_device/sysbuild/mcuboot/boards/nrf52840dk_nrf52840.overlay [hunk 10]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/ {
+	chosen {
```
Comment (nordicjm) [inline samples/sid_end_device/sysbuild/mcuboot/boards/nrf52840dk_nrf52840.overlay:8]: the app.overlay and prj.conf file from sdk-mcuboot is needed in sysbuild/mcuboot

File: samples/sid_end_device/sysbuild/mcuboot/boards/nrf5340dk_nrf5340_cpuapp.overlay [hunk 11]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+ / {
+	chosen {
+		nordic,pm-ext-flash = &mx25r64;
```
Comment (nordicjm) [inline samples/sid_end_device/sysbuild/mcuboot/boards/nrf5340dk_nrf5340_cpuapp.overlay:10]: as above

File: samples/sid_end_device/sysbuild/mcuboot/boards/thingy53_nrf5340_cpuapp.overlay [hunk 12]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+ / {
+	chosen {
+		nordic,pm-ext-flash = &mx25r64;
```
Comment (nordicjm) [inline samples/sid_end_device/sysbuild/mcuboot/boards/thingy53_nrf5340_cpuapp.overlay:10]: as above

File: samples/sid_end_device/Kconfig.sysbuild [hunk 13]
```diff
@@ -5,13 +5,13 @@
 #
 
 config NRF_DEFAULT_BLUETOOTH
-	default y if (("$(BOARD)" = "nrf5340dk_nrf5340_cpuapp") || ("$(BOARD)" = "thingy53_nrf5340_cpuapp"))
+	default y if BOARD_NRF5340DK_NRF5340_CPUAPP || BOARD_THINGY53_NRF5340_CPUAPP
```
Comment (nordicjm) [inline samples/sid_end_device/Kconfig.sysbuild:8]: squash into first commit

File: tests/manual/simple_bootloader/sysbuild/mcuboot/prj.conf [hunk 14]
```diff
@@ -0,0 +1,36 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
```
Comment (nordicjm) [inline tests/manual/simple_bootloader/sysbuild/mcuboot/prj.conf:4]: app.overlay is needed

File: tests/manual/ble/sysbuild.conf [hunk 15]
```diff
@@ -0,0 +1,7 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+SB_CONFIG_PARTITION_MANAGER=n
```
Comment (nordicjm) [inline tests/manual/ble/sysbuild.conf:7]: this seems to conflict with the above file `CONFIG_PARTITION_MANAGER_ENABLED=y`?

File: tests/manual/log/sysbuild.conf [hunk 16]
```diff
@@ -0,0 +1,7 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+SB_CONFIG_PARTITION_MANAGER=n
```
Comment (nordicjm) [inline tests/manual/log/sysbuild.conf:7]: and this

File: VERSION [hunk 17]
```diff
@@ -0,0 +1 @@
+2.6.99
```
Comment (nordicjm) [inline VERSION:1]: file can go

**Review by nordicjm — COMMENTED** [review]
commits need squashing

File: tests/manual/ble/sysbuild.conf (hunk 15 above)
Comment (ktaborowski) [reply tests/manual/ble/sysbuild.conf:7]: This is a test application, partition manager is not enabled there

File: tests/manual/log/sysbuild.conf (hunk 16 above)
Comment (ktaborowski) [reply tests/manual/log/sysbuild.conf:7]: In this test application partition manager in not enabled

File: samples/sid_end_device/Kconfig.sysbuild (hunk 7 above)
Comment (ktaborowski) [reply samples/sid_end_device/Kconfig.sysbuild]: squashed

File: samples/sid_end_device/Kconfig.sysbuild (hunk 8 above)
Comment (ktaborowski) [reply samples/sid_end_device/Kconfig.sysbuild]: squashed

File: samples/sid_end_device/sysbuild/mcuboot/boards/nrf52840dk_nrf52840.overlay (hunk 10 above)
Comment (ktaborowski) [reply samples/sid_end_device/sysbuild/mcuboot/boards/nrf52840dk_nrf52840.overlay:8]: squashed

File: samples/sid_end_device/sysbuild/mcuboot/boards/nrf5340dk_nrf5340_cpuapp.overlay (hunk 11 above)
Comment (ktaborowski) [reply samples/sid_end_device/sysbuild/mcuboot/boards/nrf5340dk_nrf5340_cpuapp.overlay:10]: squashed

File: samples/sid_end_device/sysbuild/mcuboot/boards/thingy53_nrf5340_cpuapp.overlay (hunk 12 above)
Comment (ktaborowski) [reply samples/sid_end_device/sysbuild/mcuboot/boards/thingy53_nrf5340_cpuapp.overlay:10]: squashed

File: samples/sid_end_device/Kconfig.sysbuild (hunk 13 above)
Comment (ktaborowski) [reply samples/sid_end_device/Kconfig.sysbuild:8]: squashed

File: tests/manual/simple_bootloader/sysbuild/mcuboot/prj.conf (hunk 14 above)
Comment (ktaborowski) [reply tests/manual/simple_bootloader/sysbuild/mcuboot/prj.conf:4]: added .overlay with `boot_partition`

File: samples/sid_end_device/sysbuild/mcuboot/boards/nrf52840dk_nrf52840.overlay (hunk 9 above)
Comment (ktaborowski) [reply samples/sid_end_device/sysbuild/mcuboot/boards/nrf52840dk_nrf52840.overlay:10]: squashed

**Review by totyz — APPROVED** [review]

**Review by nordicjm — APPROVED** [review]

**Pull request merged**